src/AbstractArray.php

```diff
@@ -216,30 +220,44 @@ public function __construct(array $elements = [])
     abstract public function sortKeys($order = SORT_ASC, $strategy = SORT_REGULAR);
 
     /**
-     * Apply the given function to the array elements
+     * Apply the given function to the every element of the array, collecting the results.
      *
      * @param callable $func
      *
-     * @return static The instance with modified elements
+     * @return static An array with modified elements
      */
     abstract public function map(callable $func);
 
     /**
-     * Filter array elements with given function
+     * Filter the array for elements satisfying the predicate $func.
      *
      * @param callable $func
      *
-     * @return static The instance with filtered elements
+     * @return static An array with only element satisfying $func
      */
     abstract public function filter(callable $func);
 
     /**
-     * Apply the given function to every array element
+     * Filter the array for elements not satisfying the predicate $func.
+     *
+     * @param Closure $func predicate
+     *
+     * @return $this The same array with only element not satisfying $func
+     */
+    public function filterNot(Closure $func)
```

Comment:
You better should be added it in a separate PR. But open an issue for it first would be the best way.

But anyway, I don't think we really need this method :-1: . We could easily use negation in a custom logic of closure.

So I prefer to get rid of it.